vulkaninfo/vulkaninfo.h

```diff
@@ -545,8 +525,45 @@ void freepNextChain(VkStructureHeader *first) {
     }
 }
 
+/* An ptional contains either a value or nothing. The optional asserts if a value is trying to be gotten but none exist.
+ * The interface is taken from C++17's <optional> with many aspects removed.
+ * This type assumes the optional type is 'trivial'
+ */
+namespace util {
+template <typename T>
+struct optional {
```

Comment:
The implementation itself looks alright, but to avoid confusion with the official STL version of `std::optional<T>` it might be a good idea to rename it to something like `optional_lite<T>`.

Reply:
I can change the name, though I think `trivial_optional` would likely be more suitable as it only works with trivial types.

Reply:
Fixed.